src/rgw/rgw_reshard.h

```diff
@@ -183,7 +183,14 @@ class RGWReshardWait {
   ceph::condition_variable cond;
 
   struct Waiter : boost::intrusive::list_base_hook<> {
-    boost::asio::basic_waitable_timer<Clock> timer;
+#if BOOST_VERSION < 107000
```

Comment:
Maybe put a `#warning` here so once we bump the version we can remove the conditional.